### PR TITLE
Jorgel/githooks/ktlint

### PR DESCRIPTION
**Pull request by jorg3lopez**
This PR adds ktlint format checker as part of the githooks. The ktlint format checker checks for code violations.  If the check finds code violations, the githook will not pass.

### Checks
- ./gradlew test passed
- ./gradlew testEnd2End passed
- All tasks from build passed
- ./quick-test all passed

### Testing
- [ X] Tested locally?
- [ ] Ran `quickTest all`?
- [ ] Ran `./prime test` against local Docker ReportStream container?
- [ ] Downloaded a file from `http://localhost:7071/api/download`?
- [ ] Added tests?

## Fixes
- #1963 
- #1745

### CODE REVIEW

File: prime-router/ktlint.asc
Comment (td-usds) [inline prime-router/ktlint.asc:1]: This doesn't look like it should be here.

File: prime-router/settings/organizations.yml.orig
Comment (td-usds) [inline prime-router/settings/organizations.yml.orig:1]: .orig files should (almost) never be checked in

**Comment by td-usds** [comment]
This PR is too broad and contains changes unrelated to the work being described; please rip those out and put them in a different PR

**Comment by td-usds** [comment]
Note for reviewers: the reason a bunch of `.kt` files are included in this PR is because they've been formatted. As long as they are formatting changes, it's ok; but functional changes should not be part of the PR.

File: prime-router/src/test/kotlin/ElementTests.kt
Comment (td-usds) [inline prime-router/src/test/kotlin/ElementTests.kt:471]: Why is this change in here?

**Review by td-usds — CHANGES_REQUESTED** [review]

File: prime-router/ktlint.asc
Comment (jorg3lopez) [reply prime-router/ktlint.asc:1]: good catch. This was related to installing ktlint in my system. I just deleted it form the project.

File: prime-router/settings/organizations.yml.orig
Comment (jorg3lopez) [reply prime-router/settings/organizations.yml.orig:1]: This one filtered through after the merge. I just deleted it as well.

File: prime-router/src/test/kotlin/ElementTests.kt
Comment (td-usds) [inline prime-router/src/test/kotlin/ElementTests.kt:13]: This doesn't look like the right formatting, can you validate?

File: prime-router/src/test/kotlin/ElementTests.kt
Comment (td-usds) [reply prime-router/src/test/kotlin/ElementTests.kt:472]: my question was actually: "Why is the PR that adds ktlint also adding an unrelated unit test?"

**Review by td-usds — CHANGES_REQUESTED** [review]

File: prime-router/src/main/kotlin/azure/DatabaseAccess.kt.orig
Comment (ronaldheft-usds) [inline prime-router/src/main/kotlin/azure/DatabaseAccess.kt.orig:1]: Why is this file removed? It doesn't seem to have anything to do with linting?

File: prime-router/metadata/schemas/covid-19.schema
Comment (ronaldheft-usds) [inline prime-router/metadata/schemas/covid-19.schema:1263]: Why was this line changed? It's not a kotlin file, so it shouldn't be affected by the linter.

File: prime-router/src/main/kotlin/azure/WorkflowEngine.kt.orig
Comment (ronaldheft-usds) [inline prime-router/src/main/kotlin/azure/WorkflowEngine.kt.orig:1]: Why was this file removed? It's not a kotlin file, so it shouldn't be affected by the linter.

File: prime-router/src/main/kotlin/cli/tests/Hl7Tests.kt.orig
Comment (ronaldheft-usds) [inline prime-router/src/main/kotlin/cli/tests/Hl7Tests.kt.orig:1]: Why was this file removed? It's not a kotlin file, so it shouldn't be affected by the linter.

File: prime-router/src/test/kotlin/ElementTests.kt
Comment (ronaldheft-usds) [reply prime-router/src/test/kotlin/ElementTests.kt:488]: I have the same question. This is a new test, and does not appear to be related to linting.

File: .environment/ktlint/run-ktlintCheck.sh
Comment (ronaldheft-usds) [inline .environment/ktlint/run-ktlintCheck.sh:16]: Having both options be "f" names is confusing. Could the full option be renamed to `--all` or `-a`?

File: .environment/ktlint/run-ktlintCheck.sh
Comment (ronaldheft-usds) [inline .environment/ktlint/run-ktlintCheck.sh:68]: I'm unable to run this option locally. Here is what I get:

```
prime-data-hub git:(jorgel/githooks/ktlint) .environment/ktlint/run-ktlintCheck.sh
ktlint> info: Checking for ktlint code violations...
ktlint> info: Skipping this check, there are no target files to scan.
➜  prime-data-hub git:(jorgel/githooks/ktlint) .environment/ktlint/run-ktlintCheck.sh -f
ktlint> info: Checking for ktlint code violations...
find: illegal option -- t
usage: find [-H | -L | -P] [-EXdsx] [-f path] path ... [expression]
       find [-H | -L | -P] [-EXdsx] -f path [path ...] [expression]
find: illegal option -- t
usage: find [-H | -L | -P] [-EXdsx] [-f path] path ... [expression]
       find [-H | -L | -P] [-EXdsx] -f path [path ...] [expression]
ktlint> info: Skipping this check, there are no target files to scan.
➜  prime-data-hub git:(jorgel/githooks/ktlint) .environment/ktlint/run-ktlintCheck.sh --full
ktlint> info: Checking for ktlint code violations...
find: illegal option -- t
usage: find [-H | -L | -P] [-EXdsx] [-f path] path ... [expression]
       find [-H | -L | -P] [-EXdsx] -f path [path ...] [expression]
find: illegal option -- t
usage: find [-H | -L | -P] [-EXdsx] [-f path] path ... [expression]
       find [-H | -L | -P] [-EXdsx] -f path [path ...] [expression]
ktlint> info: Skipping this check, there are no target files to scan.
```

**Review by ronaldheft-usds — CHANGES_REQUESTED** [review]
I was unable to run this locally with the --full option.

**Comment by ronaldheft-usds** [comment]
Added on additional reviews, since we should verify this works across multiple developer machines, since it will be part of the standard commit hook.

File: prime-router/src/test/kotlin/ElementTests.kt
Comment (jorg3lopez) [reply prime-router/src/test/kotlin/ElementTests.kt]: Test case belongs to another branch. I'm deleting it from this one.

File: prime-router/src/main/kotlin/azure/DatabaseAccess.kt.orig
Comment (jorg3lopez) [reply prime-router/src/main/kotlin/azure/DatabaseAccess.kt.orig:1]: .orig files are generated during a merge. These were generated when I updated my master branch and then merged it to this one.

File: prime-router/src/main/kotlin/azure/WorkflowEngine.kt.orig
Comment (jorg3lopez) [reply prime-router/src/main/kotlin/azure/WorkflowEngine.kt.orig:1]: .orig file, created during merge from master.

File: prime-router/src/main/kotlin/cli/tests/Hl7Tests.kt.orig
Comment (jorg3lopez) [reply prime-router/src/main/kotlin/cli/tests/Hl7Tests.kt.orig:1]: same as the other .orig files

File: .environment/ktlint/run-ktlintCheck.sh
Comment (jorg3lopez) [reply .environment/ktlint/run-ktlintCheck.sh:68]: Good catch, let me look into it.

File: prime-router/metadata/schemas/covid-19.schema
Comment (jorg3lopez) [reply prime-router/metadata/schemas/covid-19.schema:1263]: Leaving it as it was before.

Comment (jorg3lopez) [reply prime-router/metadata/schemas/covid-19.schema:1263]: Checking master, it doesn't have that character "-"

File: .environment/ktlint/run-ktlintCheck.sh
Comment (jorg3lopez) [reply .environment/ktlint/run-ktlintCheck.sh]: Changed line 65 and 66 to :
`$(find ./ -type f -iname "*.kt")
            $(find ./ -type f -iname "*.kts")`

Comment (jorg3lopez) [reply .environment/ktlint/run-ktlintCheck.sh]: Can you verify this works in your environment, thanks.

File: .environment/ktlint/run-ktlintCheck.sh
Comment (jorg3lopez) [reply .environment/ktlint/run-ktlintCheck.sh]: Good observation. I changed it to --all and -a

File: .environment/ktlint/run-ktlintCheck.sh
Comment (ronaldheft-usds) [reply .environment/ktlint/run-ktlintCheck.sh]: This is working now.

**Review by ronaldheft-usds — CHANGES_REQUESTED** [review]
Looking better! I can now successfully run:
```
.environment/ktlint/run-ktlintCheck.sh --all 
```

However, when doing so, I have violations. So, I'm attempting to fix them with:
```
.environment/ktlint/run-ktlintCheck.sh --all --fix
```

But it appears I cannot use both options together.

**Comment by jorg3lopez** [comment]
> Looking better! I can now successfully run:
> 
> ```
> .environment/ktlint/run-ktlintCheck.sh --all 
> ```
> 
> However, when doing so, I have violations. So, I'm attempting to fix them with:
> 
> ```
> .environment/ktlint/run-ktlintCheck.sh --all --fix
> ```
> 
> But it appears I cannot use both options together.

Nice, I'm glad we are making progress! The --all option checks the whole project for code violations. Unfortunately, the format option only applies to staged files ready to be committed. Formatting for the whole project might be something that we can add to it once the githook is up and running, let me know what you think.

**Comment by jorg3lopez** [comment]
> Looking better! I can now successfully run:
> 
> ```
> .environment/ktlint/run-ktlintCheck.sh --all 
> ```
> 
> However, when doing so, I have violations. So, I'm attempting to fix them with:
> 
> ```
> .environment/ktlint/run-ktlintCheck.sh --all --fix
> ```
> 
> But it appears I cannot use both options together.

I'm glad that it's coming together. Please try both options again, I made a couple of adjustments. As a reminder, --format only formats the staged files, --all --format will format all files in the project with .kt and kts extension.

**Comment by ronaldheft-usds** [comment]
> > Looking better! I can now successfully run:
> > ```
> > .environment/ktlint/run-ktlintCheck.sh --all 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > However, when doing so, I have violations. So, I'm attempting to fix them with:
> > ```
> > .environment/ktlint/run-ktlintCheck.sh --all --fix
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But it appears I cannot use both options together.
> 
> I'm glad that it's coming together. Please try both options again, I made a couple of adjustments. As a reminder, --format only formats the staged files, --all --format will format all files in the project with .kt and kts extension.

Just gave that a go, but still no luck:

```
 prime-data-hub git:(jorgel/githooks/ktlint) ✗ .environment/ktlint/run-ktlintCheck.sh --all         
ktlint> info: Checking for ktlint code violations...
[+] Building 1.7s (17/17) FINISHED                                                                                                                                                                                                                                                                                      
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                                                               0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                                                                                                                                0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                  0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                                                                    0.0s
 => [internal] load metadata for docker.io/library/java:alpine                                                                                                                                                                                                                                                     1.6s
 => [internal] load metadata for docker.io/library/alpine:3                                                                                                                                                                                                                                                        1.6s
 => [downloader 1/8] FROM docker.io/library/alpine:3@sha256:e1c082e3d3c45cccac829840a25941e679c25d438cc8412c2fa221cf1a824e6a                                                                                                                                                                                       0.0s
 => [stage-1 1/4] FROM docker.io/library/java:alpine@sha256:d49bf8c44670834d3dade17f8b84d709e7db47f1887f671a0e098bafa9bae49f                                                                                                                                                                                       0.0s
 => CACHED [downloader 2/8] RUN apk update     && apk add curl gnupg                                                                                                                                                                                                                                               0.0s
 => CACHED [downloader 3/8] WORKDIR /tmp                                                                                                                                                                                                                                                                           0.0s
 => CACHED [downloader 4/8] RUN curl -sS https://keybase.io/ktlint/pgp_keys.asc | gpg --import                                                                                                                                                                                                                     0.0s
 => CACHED [downloader 5/8] RUN curl --output /tmp/ktlint -sSLO https://github.com/pinterest/ktlint/releases/download/0.42.1/ktlint                                                                                                                                                                                0.0s
 => CACHED [downloader 6/8] RUN curl --output /tmp/ktlint.asc -sSLO https://github.com/pinterest/ktlint/releases/download/0.42.1/ktlint.asc                                                                                                                                                                        0.0s
 => CACHED [downloader 7/8] RUN gpg --verify ktlint.asc                                                                                                                                                                                                                                                            0.0s
 => CACHED [downloader 8/8] RUN chmod a+x /tmp/ktlint                                                                                                                                                                                                                                                              0.0s
 => CACHED [stage-1 2/4] COPY --from=downloader /tmp/ktlint /usr/bin/                                                                                                                                                                                                                                              0.0s
 => CACHED [stage-1 3/4] RUN mkdir "/src"                                                                                                                                                                                                                                                                          0.0s
 => CACHED [stage-1 4/4] WORKDIR /src                                                                                                                                                                                                                                                                              0.0s
 => exporting to image                                                                                                                                                                                                                                                                                             0.0s
 => => exporting layers                                                                                                                                                                                                                                                                                            0.0s
 => => writing image sha256:6b00e8e9750ffda1f364a2fca8dccf1f82c1985e0511307d628eff8c8b7974df                                                                                                                                                                                                                       0.0s
 => => naming to docker.io/library/reportstream-ktlint:latest                                                                                                                                                                                                                                                      0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
    ././prototypes/as2./main/kotlin/AS2ExpCommand.kt:17:1: Wildcard import (cannot be auto-corrected)
    ././prototypes/as2./main/kotlin/AS2ExpCommand.kt:19:1: Needless blank line(s)
    ././prototypes/as2./main/kotlin/AS2ExpCommand.kt:50:20: Missing spacing before ":"
    ././prototypes/as2./main/kotlin/AS2ExpCommand.kt:52:44: Missing spacing around "="
    ././prototypes/as2./main/kotlin/AS2ExpCommand.kt:53:46: Missing spacing around "="
    ././prototypes/as2./main/kotlin/AS2ExpCommand.kt:54:44: Missing spacing around "="
    ././prototypes/as2./main/kotlin/AS2ExpCommand.kt:63:31: Unexpected spacing before "("
    ././prototypes/as2./main/kotlin/AS2ExpCommand.kt:79:1: Unexpected indentation (16) (should be 12)
    ././prototypes/as2./main/kotlin/AS2ExpCommand.kt:80:1: Unexpected indentation (16) (should be 12)
    ././prototypes/as2./main/kotlin/AS2ExpCommand.kt:89:42: Redundant "toString()" call in string template
    ././prototypes/as2/build.gradle.kts:22:2: Redundant newline (\n) at the end of file
    ././prototypes/as2/settings.gradle.kts:1:28: Redundant newline (\n) at the end of file
    ././prototypes/as2/settings.gradle.kts:3:1: Needless blank line(s)
ktlint> ERROR: You likely have ktlint violations, check the output or 'ktlint.log' for more information
ktlint> info: You may be able to fix the violations by invoking '.environment/ktlint/run-ktlintCheck.sh --format'.
```


```
➜  prime-data-hub git:(jorgel/githooks/ktlint) .environment/ktlint/run-ktlintCheck.sh --all --format
ktlint> info: Checking for ktlint code violations...
[+] Building 0.6s (17/17) FINISHED                                                                                                                                                                                                                                                                                      
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                                                               0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                                                                                                                                0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                  0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                                                                    0.0s
 => [internal] load metadata for docker.io/library/java:alpine                                                                                                                                                                                                                                                     0.3s
 => [internal] load metadata for docker.io/library/alpine:3                                                                                                                                                                                                                                                        0.4s
 => [downloader 1/8] FROM docker.io/library/alpine:3@sha256:e1c082e3d3c45cccac829840a25941e679c25d438cc8412c2fa221cf1a824e6a                                                                                                                                                                                       0.0s
 => [stage-1 1/4] FROM docker.io/library/java:alpine@sha256:d49bf8c44670834d3dade17f8b84d709e7db47f1887f671a0e098bafa9bae49f                                                                                                                                                                                       0.0s
 => CACHED [downloader 2/8] RUN apk update     && apk add curl gnupg                                                                                                                                                                                                                                               0.0s
 => CACHED [downloader 3/8] WORKDIR /tmp                                                                                                                                                                                                                                                                           0.0s
 => CACHED [downloader 4/8] RUN curl -sS https://keybase.io/ktlint/pgp_keys.asc | gpg --import                                                                                                                                                                                                                     0.0s
 => CACHED [downloader 5/8] RUN curl --output /tmp/ktlint -sSLO https://github.com/pinterest/ktlint/releases/download/0.42.1/ktlint                                                                                                                                                                                0.0s
 => CACHED [downloader 6/8] RUN curl --output /tmp/ktlint.asc -sSLO https://github.com/pinterest/ktlint/releases/download/0.42.1/ktlint.asc                                                                                                                                                                        0.0s
 => CACHED [downloader 7/8] RUN gpg --verify ktlint.asc                                                                                                                                                                                                                                                            0.0s
 => CACHED [downloader 8/8] RUN chmod a+x /tmp/ktlint                                                                                                                                                                                                                                                              0.0s
 => CACHED [stage-1 2/4] COPY --from=downloader /tmp/ktlint /usr/bin/                                                                                                                                                                                                                                              0.0s
 => CACHED [stage-1 3/4] RUN mkdir "/src"                                                                                                                                                                                                                                                                          0.0s
 => CACHED [stage-1 4/4] WORKDIR /src                                                                                                                                                                                                                                                                              0.0s
 => exporting to image                                                                                                                                                                                                                                                                                             0.0s
 => => exporting layers                                                                                                                                                                                                                                                                                            0.0s
 => => writing image sha256:6b00e8e9750ffda1f364a2fca8dccf1f82c1985e0511307d628eff8c8b7974df                                                                                                                                                                                                                       0.0s
 => => naming to docker.io/library/reportstream-ktlint:latest                                                                                                                                                                                                                                                      0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
    ././prototypes/as2./main/kotlin/AS2ExpCommand.kt:17:1: Wildcard import (cannot be auto-corrected)
ktlint> ERROR: You likely have ktlint violations, check the output or 'ktlint.log' for more information
ktlint> info: You may be able to fix the violations by invoking '.environment/ktlint/run-ktlintCheck.sh --format'.
```

```
➜  prime-data-hub git:(jorgel/githooks/ktlint) ✗ .environment/ktlint/run-ktlintCheck.sh --all         
ktlint> info: Checking for ktlint code violations...
[+] Building 0.5s (17/17) FINISHED                                                                                                                                                                                                                                                                                      
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                                                               0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                                                                                                                                0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                  0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                                                                    0.0s
 => [internal] load metadata for docker.io/library/java:alpine                                                                                                                                                                                                                                                     0.3s
 => [internal] load metadata for docker.io/library/alpine:3                                                                                                                                                                                                                                                        0.3s
 => [downloader 1/8] FROM docker.io/library/alpine:3@sha256:e1c082e3d3c45cccac829840a25941e679c25d438cc8412c2fa221cf1a824e6a                                                                                                                                                                                       0.0s
 => [stage-1 1/4] FROM docker.io/library/java:alpine@sha256:d49bf8c44670834d3dade17f8b84d709e7db47f1887f671a0e098bafa9bae49f                                                                                                                                                                                       0.0s
 => CACHED [downloader 2/8] RUN apk update     && apk add curl gnupg                                                                                                                                                                                                                                               0.0s
 => CACHED [downloader 3/8] WORKDIR /tmp                                                                                                                                                                                                                                                                           0.0s
 => CACHED [downloader 4/8] RUN curl -sS https://keybase.io/ktlint/pgp_keys.asc | gpg --import                                                                                                                                                                                                                     0.0s
 => CACHED [downloader 5/8] RUN curl --output /tmp/ktlint -sSLO https://github.com/pinterest/ktlint/releases/download/0.42.1/ktlint                                                                                                                                                                                0.0s
 => CACHED [downloader 6/8] RUN curl --output /tmp/ktlint.asc -sSLO https://github.com/pinterest/ktlint/releases/download/0.42.1/ktlint.asc                                                                                                                                                                        0.0s
 => CACHED [downloader 7/8] RUN gpg --verify ktlint.asc                                                                                                                                                                                                                                                            0.0s
 => CACHED [downloader 8/8] RUN chmod a+x /tmp/ktlint                                                                                                                                                                                                                                                              0.0s
 => CACHED [stage-1 2/4] COPY --from=downloader /tmp/ktlint /usr/bin/                                                                                                                                                                                                                                              0.0s
 => CACHED [stage-1 3/4] RUN mkdir "/src"                                                                                                                                                                                                                                                                          0.0s
 => CACHED [stage-1 4/4] WORKDIR /src                                                                                                                                                                                                                                                                              0.0s
 => exporting to image                                                                                                                                                                                                                                                                                             0.0s
 => => exporting layers                                                                                                                                                                                                                                                                                            0.0s
 => => writing image sha256:6b00e8e9750ffda1f364a2fca8dccf1f82c1985e0511307d628eff8c8b7974df                                                                                                                                                                                                                       0.0s
 => => naming to docker.io/library/reportstream-ktlint:latest                                                                                                                                                                                                                                                      0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
    ././prototypes/as2./main/kotlin/AS2ExpCommand.kt:17:1: Wildcard import (cannot be auto-corrected)
ktlint> ERROR: You likely have ktlint violations, check the output or 'ktlint.log' for more information
ktlint> info: You may be able to fix the violations by invoking '.environment/ktlint/run-ktlintCheck.sh --format'.
```

**Review by ronaldheft-usds — APPROVED** [review]
Talked to @jorg3lopez over Slack, and the issue was some fixes could not be auto-formatted. All looks good on my end.

I am approving, but we should get additional approvals from someone running on Windows and someone on Linux before merging.

**Pull request merged**
